Source/Core/VideoCommon/CPMemory.cpp

```diff
@@ -17,14 +18,9 @@ CPState g_preprocess_cp_state;
 
 void CopyPreprocessCPStateFromMain()
 {
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wclass-memaccess"
-#endif
-  std::memcpy(&g_preprocess_cp_state, &g_main_cp_state, sizeof(CPState));
-#ifdef __GNUC__
-#pragma GCC diagnostic pop
-#endif
+  static_assert(std::is_trivially_copyable_v<CPState>);
+  std::memcpy(static_cast<void*>(&g_preprocess_cp_state),
+              static_cast<const void*>(&g_main_cp_state), sizeof(CPState));
```

Comment:
Is there a reason why we can't just write `g_preprocess_cp_state = g_main_cp_state`? I feel like that should be possible (although I tried it in the past and it didn't work then and I don't remember why).

Reply:
oh, right, the same as above: `BitField` isn't copy-asignable. This was avoided in the fifo analyzer because it uses copy construction instead of copy assignment (`AnalyzedFrameInfo`'s `AddPart` uses `parts.emplace_back` which calls the `FramePart` constuctor).

I further confused myself by looking at `BitField.h`, seeing `BitField& operator=(const BitField&) = delete;`, and somehow concluding that copy assignment was still allowed and it was copy construction that was disabled. I'm not sure how I managed that...

In any case, this code is good for now.